src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -1431,24 +1460,138 @@ public void doFinishLogin(StaplerRequest2 request, StaplerResponse2 response) th
+                // the rest of chain be processed
+                return Optional.of(true);
+            } // else do nothing and continue evaluating this request
+        }
+        return Optional.empty();
+    }
+
+    @VisibleForTesting
+    Optional<Boolean> attemptBearerToken(HttpServletRequest httpRequest) {
+        if (!isAllowJWTBearerTokenAccess()) {
+            return Optional.empty();
+        }
+
+        // check if this is a valid Bearer token based request
+        String authHeader = httpRequest.getHeader("Authorization");
+        if (authHeader != null && authHeader.startsWith("Bearer ")) {
+            try {
+                JWT jwt = JWTParser.parse(authHeader.substring(7));
+                OIDCProviderMetadata metadata = serverConfiguration.toProviderMetadata();
+                filterNonFIPS140CompliantAlgorithms(metadata);
+
+                ConfigurableJWTProcessor<SecurityContext> jwtProcessor = new DefaultJWTProcessor<>();
+                if (isDisableTokenVerification()) {
+                    // nimbus (probably for good reason) by default rejects all unsigned jwt and forces us to
+                    // override this method if we want to allow it regardless of recommendations
+                    jwtProcessor = new DefaultJWTProcessor<>() {
+                        @Override
+                        public JWTClaimsSet process(PlainJWT plainJWT, SecurityContext context)
+                                throws BadJOSEException, JOSEException {
+                            if (getJWSTypeVerifier() == null) {
+                                throw new BadJOSEException(
+                                        "Plain JWT rejected: No JWS header typ (type) verifier is configured");
+                            }
+                            getJWSTypeVerifier().verify(plainJWT.getHeader().getType(), context);
+                            JWTClaimsSet claimsSet = extractJWTClaimsSet(plainJWT);
+                            return verifyJWTClaimsSet(claimsSet, context);
+                        }
+                    };
+                } else {
+                    var jwk = JWKSourceBuilder.create(metadata.getJWKSetURI().toURL())
+                            .build();
+                    var keySelector = new JWSVerificationKeySelector<>(Set.copyOf(metadata.getIDTokenJWSAlgs()), jwk);
+                    jwtProcessor.setJWSKeySelector(keySelector);
+                }
+                jwtProcessor.setJWSTypeVerifier(new DefaultJOSEObjectTypeVerifier<>(JOSEObjectType.JWT));
+
+                var exactMatchClaims = new JWTClaimsSet.Builder()
+                        .issuer(metadata.getIssuer().getValue())
+                        .build();
+                var requiredClaims = Set.of(JWTClaimNames.EXPIRATION_TIME, userNameField);
+                String requiredAudience = clientId;
+                var verifier = new DefaultJWTClaimsVerifier<>(requiredAudience, exactMatchClaims, requiredClaims) {
```

Comment:
If an unrelated plugin is accepting `Authorization: Bearer $somejwt` for an unrelated HTTP endpoint using an unrelated audience, and this interceptor happens to run, will it fail? Will it log an alarming stack trace? Preferably it should quietly ignore unrelated JWTs.

Reply:
I don't think it does support that and tbh I also don't think we need to. I think all validations issues have the same "severity" so I'd rather not distinguish them